### DISCUSSION TITLE
[Bug?] Wrong commit used when pushing tag

### DISCUSSION THREAD

**Discussion by haydenbaker** [jreleaser/jreleaser#1676] [Q&A]
When executing a full release using JReleaser, we had an issue where the release tag was set on the wrong commit. Here's the full scenario (JReleaser config [here](https://github.com/smithy-lang/smithy/blob/main/smithy-cli/build.gradle#L266-L356)):

1. We create a version-bump commit where we update a VERSION file ([example](https://github.com/smithy-lang/smithy/commit/e8e104e890971bb7f0abba5a2319f6e58127c49a))
2. We checkout that specific commit before running the release
3. We run a full-release (with tagging enabled), which creates the GitHub release with a new tag

This time around, another PR was merged between the times of step (1) and (2), but this shouldn't matter since we checkout the exact commit we want to release with. The logs even confirm that the release used the [right commit](https://github.com/smithy-lang/smithy/commit/e8e104e890971bb7f0abba5a2319f6e58127c49a):
```
[INFO]  HEAD is at e8e104e
```

However, when the tag was pushed, it referenced the [newer commit](https://github.com/smithy-lang/smithy/commit/1249c68d1bcae64c7008b5b8cb22b502570ec9fd) instead of the commit we had checked out. We have since rectified the tag (1.50.0) to use the right commit, but it was strange. The trace log is [here](https://gist.github.com/haydenbaker/a826323865f4299e57d51da4d49d6ed7).

Any thoughts?

**Comment by haydenbaker** [comment]
FWIW, I don't believe we've ran into this issue in the past with similar cases (new commits added between the steps).

**Reply by aalmiray** [reply]
This sounds like a bug to me. Based on the following code where a branch mismatched would result in a fresh checkout without taking the current commit hash

https://github.com/jreleaser/jreleaser/blob/3b180e832b83465167fa43628c8388a30637e847/sdks/jreleaser-github-java-sdk/src/main/java/org/jreleaser/sdk/github/GithubReleaser.java#L172-L177

**Reply by haydenbaker** [reply]
I see. It must've just been coincidence then in the past that we didn't encounter this. Thanks so much for triaging!

Do you want me to create an issue from this?

**Reply by haydenbaker** [reply]
https://github.com/jreleaser/jreleaser/issues/1692
